src/main/java/org/elasticsearch/index/search/geo/GeoPolygonFilter.java

```diff
@@ -56,8 +76,18 @@ public String fieldName() {
 
     @Override
     public DocIdSet getDocIdSet(LeafReaderContext context, Bits acceptedDocs) throws IOException {
-        final MultiGeoPointValues values = indexFieldData.load(context).getGeoPointValues();
-        return new GeoPolygonDocIdSet(context.reader().maxDoc(), acceptedDocs, values, points);
+        MultiGeoPointValues values = indexFieldData.load(context).getGeoPointValues();
+        GeoPolygonDocIdSet polygonDocSet = new GeoPolygonDocIdSet(context.reader().maxDoc(), acceptedDocs, values, points);
+
+        if (boundingBoxFilter == null) {
+            return polygonDocSet;
+        } else {
+            DocIdSet boundingBoxDocSet = boundingBoxFilter.getDocIdSet(context, acceptedDocs);
+            if (DocIdSets.isEmpty(boundingBoxDocSet)) { // Necessary to avoid a NullPointerException.
```

Comment:
We should be returning `null` in this case. If nothing matches the bounding box, nothing can match the polygon.